docs/en/concepts-and-designs/lal.md

```diff
@@ -0,0 +1,293 @@
+    }
+    // ...
+}
+```
+
+- `grok`
+
+<!-- TODO: grok Java library has poor performance, need to benchmark it, the idea is basically the same with `regexp` above -->
+
+### Extractor
+
+Extractors aim to extract metadata from the logs. The metadata can be a service name, a service instance name, an
+endpoint name, or even a trace ID, all of which can be associated with the existing traces and metrics.
+
+- `service`
+
+`service` extracts the service name from the `parsed` result, and set it into the `LogData`, which will be persisted (if
+not dropped) and is used to associate with traces / metrics.
+
+- `instance`
+
+`instance` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `endpoint`
+
+`endpoint` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `traceId`
+
+`traceId` extracts the trace ID from the `parsed` result, and set it into the `LogData`, which will be persisted (if not
+dropped) and is used to associate with traces / metrics.
+
+- `segmentId`
+
+`segmentId` extracts the segment ID from the `parsed` result, and set it into the `LogData`, which will be persisted (if
+not dropped) and is used to associate with traces / metrics.
+
+- `spanId`
+
+`spanId` extracts the span ID from the `parsed` result, and set it into the `LogData`, which will be persisted (if not
+dropped) and is used to associate with traces / metrics.
+
+- `timestamp`
+
+`timestamp` extracts the timestamp from the `parsed` result, and set it into the `LogData`, which will be persisted (if
+not dropped) and is used to associate with traces / metrics.
+
+- `tag`
+
+`tag` extracts the tags from the `parsed` result, and set them into the `LogData`. The form of this extractor is
+something like `tag key1: value, key2: value2`, you can use the properties of `parsed` as both keys and values.
+
+```groovy
+filter {
+    // ... parser
+
+    extractor {
+        tag level: parsed.level, (parsed.statusCode): parsed.statusMsg
+        tag anotherKey: "anotherConstantValue"
+    }
+}
+```
+
+- `metrics`
+
+`metrics` extracts / generates metrics from the logs, and sends the generated metrics to the meter system, you can
+configure [MAL](mal.md) for further analysis of these metrics. The dedicated MAL config files are under
+directory `log-mal-rules`, you can set `log-analyzer/default/malFiles` to enable configured files.
+
+```yaml
+# application.yml
+# ...
+log-analyzer:
+  selector: ${SW_LOG_ANALYZER:default}
+  default:
+    lalFiles: ${SW_LOG_LAL_FILES:my-lal-config.yaml} # files are under "lal" directory
+    malFiles: ${SW_LOG_MAL_FILES:my-lal-mal-config.yaml,another-lal-mal-config.yaml} # files are under "log-mal-rules" directory
```

Comment:
The rule files are not including suffix, please recheck and keep consistent.

Reply:
> The rule files are not including suffix, please recheck and keep consistent.

Actually they are inconsistent already, `spring-sleuth.yaml`, `self`, and `istio-controlplane`, which style should we adopt?

Reply:
Then it is a meter system issue, we should not include suffix there.